src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1504,7 +1504,7 @@ private static bool VectorContainsNonAsciiChar(Vector128<byte> asciiVector)
             // prefer architecture specific intrinsic as they offer better perf
             if (Sse41.IsSupported)
             {
-                return !Sse41.TestZ(asciiVector, Vector128.Create((byte)0x80));
+                return (asciiVector & Vector128.Create((byte)0x80)) != Vector128<byte>.Zero;
```

Comment:
nit: Can we make this general method do:
```csharp
            if (AdvSimd.Arm64.IsSupported)
            {
                Vector128<byte> maxBytes = AdvSimd.Arm64.MaxPairwise(asciiVector, asciiVector);
                return (maxBytes.AsUInt64().ToScalar() & 0x8080808080808080) != 0;
            }
            else
            {
                return (asciiVector & Vector128.Create<byte>(0x80)) != Vector128<byte>.Zero;
            }
```

I'd like to eventually get rid of the Arm path as well, we just don't have the pattern recognition in place atm.

Reply:
> nit: Can we make this general method do:
> 
> ```cs
>             if (AdvSimd.Arm64.IsSupported)
>             {
>                 Vector128<byte> maxBytes = AdvSimd.Arm64.MaxPairwise(asciiVector, asciiVector);
>                 return (maxBytes.AsUInt64().ToScalar() & 0x8080808080808080) != 0;
>             }
>             else
>             {
>                 return (asciiVector & Vector128.Create<byte>(0x80)) != Vector128<byte>.Zero;
>             }
> ```
> 
> I'd like to eventually get rid of the Arm path as well, we just don't have the pattern recognition in place atm.

we have, it's https://github.com/dotnet/runtime/pull/102705 🙂 just need to finish it (hopefully, today)

Reply:
These changes can be made in a follow-up PR once https://github.com/dotnet/runtime/pull/102705 is merged.